mmdet/models/utils/position_encoding.py

```diff
@@ -0,0 +1,147 @@
+import math
+
+import torch
+import torch.nn as nn
+from mmcv.cnn import uniform_init
+
+from .builder import POSITION_ENCODING
+
+
+@POSITION_ENCODING.register_module()
+class SinePositionEmbedding(nn.Module):
+    """Position encoding with sine and cosine functions.
+
+    See `End-to-End Object Detection with Transformers
+    <https://arxiv.org/pdf/2005.12872>`_ for details.
+
+    Args:
+        num_pos_feats (int): The feature dimension for each position
+            along x-axis or y-axis. Note the final returned dimension
+            for each position is 2 times of this value.
+        temperature (int): The temperature used for calculating the
+            position embedding. Default 10000.
+        normalize (bool): Whether to do normalization when calculating
+            the position embedding. Default False.
+        scale (float): A scale factor for calculating the position
```

Comment:
A scale factor that scales the position embedding. The `scale` will be used only when `normalize=True`. Default 2*pi.